server/services/applicationService.ts

```diff
@@ -103,6 +103,15 @@ export default class ApplicationService {
     return applicationData
   }
 
+  private removeConditionalData(applicationData: AnyValue): AnyValue {
```

Comment:
these changes definitely help! 

Would this be clearer if the function name reflected the fact that we're only clearing "conditional" answers where their follow-on question is no longer applicable? e.g.

`removeStaleFollowOnAnswers()` or
`clearOrphanedFollowOnAnswers()`

Reply:
good point... how do you feel about `deleteOrphanedFollowOnAnswers`? https://github.com/ministryofjustice/hmpps-community-accommodation-tier-2-ui/pull/284#discussion_r1384706514